@sergioifg94  I have tested your pr on our rhmi-cssre staging cluster. Once I rebuilt the go binary it worked for a time but bailed out on the with an error see below. A suggestion might be to search all openshift-* namespaces as it does really need to check all namespace on a given cluster.

```
./cli
2020/04/20 10:03:03 generating report for all namespaces
checking image : registry.access.redhat.com/rhscl/httpd-24-rhel7:latest
skipping tag  latest  as it does not match on major minor patch version 2.4.*
skipping tag  2.4  as it does not match on major minor patch version 2.4.*
skipping tag  2.4  as it does not match on major minor patch version 2.4.*
skipping tag  latest  as it does not match on major minor patch version 2.4.*
skipping tag  2.4  as it does not match on major minor patch version 2.4.*
skipping tag  latest  as it does not match on major minor patch version 2.4.*
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT                     | IMAGE                | IMAGE HASH                                                       | IMAGE STREAM | TAG    | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| web-console-config-web-server | rhscl/httpd-24-rhel7 | d726ce74164b1b757c5f6523c480ae0a84fdd09ae5ad8e2252a8e185c8f9273e | false        | latest | false              | 2.4-109              | 2.4-110          | latest       | true               | false                       |             0 |              0 |             0 |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT                     | IMAGE                | IMAGE HASH                                                       | IMAGE STREAM | TAG    | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| web-console-config-web-server | rhscl/httpd-24-rhel7 | d726ce74164b1b757c5f6523c480ae0a84fdd09ae5ad8e2252a8e185c8f9273e | false        | latest | false              | 2.4-109              | 2.4-110          | latest       | true               | false                       |             0 |              0 |             0 |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT                     | IMAGE                | IMAGE HASH                                                       | IMAGE STREAM | TAG    | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| web-console-config-web-server | rhscl/httpd-24-rhel7 | d726ce74164b1b757c5f6523c480ae0a84fdd09ae5ad8e2252a8e185c8f9273e | false        | latest | false              | 2.4-109              | 2.4-110          | latest       | true               | false                       |             0 |              0 |             0 |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT                     | IMAGE                | IMAGE HASH                                                       | IMAGE STREAM | TAG    | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| web-console-config-web-server | rhscl/httpd-24-rhel7 | d726ce74164b1b757c5f6523c480ae0a84fdd09ae5ad8e2252a8e185c8f9273e | false        | latest | false              | 2.4-109              | 2.4-110          | latest       | true               | false                       |             0 |              0 |             0 |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/integr8ly/heimdall/pkg/cluster.ParseImage(0xc0006e2280, 0x3e, 0xc0004145a0)
	/Users/markfreer/repos/heimdall/pkg/cluster/images.go:191 +0x5b0
github.com/integr8ly/heimdall/pkg/cluster.(*ImageService).FindImagesFromLabels(0xc0002dea40, 0xc000423399, 0x7, 0xc00062b0e0, 0x2, 0x2, 0x33, 0x0, 0x0)
	/Users/markfreer/repos/heimdall/pkg/cluster/images.go:149 +0xeec
github.com/integr8ly/heimdall/pkg/controller/deploymentconfigs.(*Reports).GetImages(0xc000545c00, 0xc0005453f0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/markfreer/repos/heimdall/pkg/controller/deploymentconfigs/report.go:109 +0x49e
github.com/integr8ly/heimdall/pkg/controller/deploymentconfigs.(*Reports).Generate(0xc000545c00, 0xc000404a84, 0x7, 0x1fd2299, 0x1, 0x0, 0x0, 0xc00072f930, 0x7, 0xc00072f870)
	/Users/markfreer/repos/heimdall/pkg/controller/deploymentconfigs/report.go:60 +0xb09
main.main()
	/Users/markfreer/repos/heimdall/cmd/cli/main.go:57 +0xc63

```
@sergioifg94  This is work for no input and the new option we suggest of all openshift-* perfix namespaces.


```

/cli
checking image : registry.access.redhat.com/rhscl/httpd-24-rhel7:latest
skipping tag  latest  as it does not match on major minor patch version 2.4.*
skipping tag  2.4  as it does not match on major minor patch version 2.4.*
skipping tag  2.4  as it does not match on major minor patch version 2.4.*
skipping tag  latest  as it does not match on major minor patch version 2.4.*
skipping tag  2.4  as it does not match on major minor patch version 2.4.*
skipping tag  latest  as it does not match on major minor patch version 2.4.*
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT                     | IMAGE                | IMAGE HASH                                                       | IMAGE STREAM | TAG    | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| web-console-config-web-server | rhscl/httpd-24-rhel7 | d726ce74164b1b757c5f6523c480ae0a84fdd09ae5ad8e2252a8e185c8f9273e | false        | latest | false              | 2.4-109              | 2.4-110          | latest       | true               | false                       |             0 |              0 |             0 |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT                     | IMAGE                | IMAGE HASH                                                       | IMAGE STREAM | TAG    | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| web-console-config-web-server | rhscl/httpd-24-rhel7 | d726ce74164b1b757c5f6523c480ae0a84fdd09ae5ad8e2252a8e185c8f9273e | false        | latest | false              | 2.4-109              | 2.4-110          | latest       | true               | false                       |             0 |              0 |             0 |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT                     | IMAGE                | IMAGE HASH                                                       | IMAGE STREAM | TAG    | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| web-console-config-web-server | rhscl/httpd-24-rhel7 | d726ce74164b1b757c5f6523c480ae0a84fdd09ae5ad8e2252a8e185c8f9273e | false        | latest | false              | 2.4-109              | 2.4-110          | latest       | true               | false                       |             0 |              0 |             0 |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT                     | IMAGE                | IMAGE HASH                                                       | IMAGE STREAM | TAG    | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| web-console-config-web-server | rhscl/httpd-24-rhel7 | d726ce74164b1b757c5f6523c480ae0a84fdd09ae5ad8e2252a8e185c8f9273e | false        | latest | false              | 2.4-109              | 2.4-110          | latest       | true               | false                       |             0 |              0 |             0 |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT                     | IMAGE                | IMAGE HASH                                                       | IMAGE STREAM | TAG    | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| web-console-config-web-server | rhscl/httpd-24-rhel7 | d726ce74164b1b757c5f6523c480ae0a84fdd09ae5ad8e2252a8e185c8f9273e | false        | latest | false              | 2.4-109              | 2.4-110          | latest       | true               | false                       |             0 |              0 |             0 |
+-------------------------------+----------------------+------------------------------------------------------------------+--------------+--------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+


```
The second option with the openshift-* prefix is also working  :

```
./cli -namespace-pattern openshift-.*
checking image : registry.redhat.io/3scale-amp2/apicast-gateway-rhel7:3scale2.7
checking image : registry.redhat.io/3scale-amp2/backend-rhel7:3scale2.7
checking image : registry.redhat.io/rhscl/redis-32-rhel7:3.2
skipping tag  latest  as it does not match on major minor patch version 3.2.*
checking image : registry.redhat.io/3scale-amp2/system-rhel7:3scale2.7
checking image : registry.redhat.io/3scale-amp2/memcached-rhel7:3scale2.7
checking image : registry.redhat.io/rhscl/mysql-57-rhel7:5.7
skipping tag  5.7  as it does not match on major minor patch version 5.7.*
skipping tag  latest  as it does not match on major minor patch version 5.7.*
checking image : registry.redhat.io/3scale-amp2/zync-rhel7:3scale2.7
checking image : registry.redhat.io/rhscl/postgresql-10-rhel7
+--------------------+-----------------------------------+------------------------------------------------------------------+--------------+-----------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT          | IMAGE                             | IMAGE HASH                                                       | IMAGE STREAM | TAG       | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |
+--------------------+-----------------------------------+------------------------------------------------------------------+--------------+-----------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| apicast-production | 3scale-amp2/apicast-gateway-rhel7 | 26700c6cb5f43fa8061cee274caef5561646995a0e7d803273f394d9cdda1391 | true         | 3scale2.7 | true               | 1.16                 | 1.16             | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| apicast-staging    | 3scale-amp2/apicast-gateway-rhel7 | 26700c6cb5f43fa8061cee274caef5561646995a0e7d803273f394d9cdda1391 | true         | 3scale2.7 | true               | 1.16                 | 1.16             | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| backend-cron       | 3scale-amp2/backend-rhel7         | d13c45ea48efd2a87cb5efb8b745b80a298cdb789e41995fbdedfcc0bfa6a5c5 | true         | 3scale2.7 | true               | 1.10                 | 1.10             | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| backend-listener   | 3scale-amp2/backend-rhel7         | d13c45ea48efd2a87cb5efb8b745b80a298cdb789e41995fbdedfcc0bfa6a5c5 | true         | 3scale2.7 | true               | 1.10                 | 1.10             | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| backend-redis      | rhscl/redis-32-rhel7              | a9bdf52384a222635efc0284db47d12fbde8c3d0fcb66517ba8eefad1d4e9dc9 | true         | 3.2       | true               | 3.2-47               | 3.2-47           | 3.2          | true               | true                        |             0 |              0 |             0 |
| backend-worker     | 3scale-amp2/backend-rhel7         | d13c45ea48efd2a87cb5efb8b745b80a298cdb789e41995fbdedfcc0bfa6a5c5 | true         | 3scale2.7 | true               | 1.10                 | 1.10             | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| system-app         | 3scale-amp2/system-rhel7          | abe1c074e975b6bf6a1f527a1820671636c393517f2f850f7299c24ae88ba2ab | true         | 3scale2.7 | true               | 3scale2.7.1          | 3scale2.7.1      | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| system-memcache    | 3scale-amp2/memcached-rhel7       | 9c897d68eebb3a704756b9ddf32358c5b99786055aa1b94cc79be4b9cd3d3f03 | true         | 3scale2.7 | true               | 3scale2.7            | 3scale2.7        | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| system-mysql       | rhscl/mysql-57-rhel7              | 9a781abe7581cc141e14a7e404ec34125b3e89c008b14f4e7b41e094fd3049fe | true         | 5.7       | true               | 5.7-49               | 5.7-49           | 5.7          | true               | true                        |             0 |              0 |             0 |
| system-redis       | rhscl/redis-32-rhel7              | a9bdf52384a222635efc0284db47d12fbde8c3d0fcb66517ba8eefad1d4e9dc9 | true         | 3.2       | true               | 3.2-47               | 3.2-47           | 3.2          | true               | true                        |             0 |              0 |             0 |
| system-sidekiq     | 3scale-amp2/system-rhel7          | abe1c074e975b6bf6a1f527a1820671636c393517f2f850f7299c24ae88ba2ab | true         | 3scale2.7 | true               | 3scale2.7.1          | 3scale2.7.1      | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| system-sphinx      | 3scale-amp2/system-rhel7          | abe1c074e975b6bf6a1f527a1820671636c393517f2f850f7299c24ae88ba2ab | true         | 3scale2.7 | true               | 3scale2.7.1          | 3scale2.7.1      | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| zync               | 3scale-amp2/zync-rhel7            | 90a18fccbc5481a1952841cf6497af3f1e5920310a7c469a26bb2d5b4032ef9f | true         | 3scale2.7 | true               | 3scale2.7.1          | 3scale2.7.1      | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
| zync-database      | rhscl/postgresql-10-rhel7         | de3ab628b403dc5eed986a7f392c34687bddafee7bdfccfd65cecf137ade3dfd | true         | latest    | false              | 1-47                 | 1-47             | latest       | true               | false                       |             0 |              0 |             0 |
| zync-que           | 3scale-amp2/zync-rhel7            | 90a18fccbc5481a1952841cf6497af3f1e5920310a7c469a26bb2d5b4032ef9f | true         | 3scale2.7 | true               | 3scale2.7.1          | 3scale2.7.1      | 3scale2.7    | true               | true                        |             0 |              0 |             0 |
+--------------------+-----------------------------------+------------------------------------------------------------------+--------------+-----------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
+--------------------+-----------------------------------+------------------------------------------------------------------+--------------+-----------+--------------------+----------------------+------------------+--------------+--------------------+-----------------------------+---------------+----------------+---------------+
| COMPONENT          | IMAGE                             | IMAGE HASH                                                       | IMAGE STREAM | TAG       | UPTO DATE WITH TAG | PERSISTENT IMAGE TAG | LATEST PATCH TAG | FLOATING TAG | USING FLOATING TAG | UPTO DATE WITH FLOATING TAG | CRITICAL CVES | IMPORTANT CVES | MODERATE CVES |

```

/lgtm

--- a/cmd/cli/main.go
+++ b/cmd/cli/main.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"regexp"
 	"strings"
 
 	"github.com/integr8ly/heimdall/pkg/cluster"
@@ -24,6 +25,7 @@ import (
 
 func main() {
 	namespacePtr := flag.String("namespaces", "", "the namespaces to check")
+	namespacePatternPtr := flag.String("namespace-pattern", "", "a go compilant regular expression to include only matching namespaces")
 	componentPtr := flag.String("component", "*", "the dc or deployment name to check in the namespace")
 	labelPodsPtr := flag.String("label-pods", "false", "add labels to the pods with the info discovered")
 	flag.Parse()
@@ -51,6 +53,10 @@ func main() {
 	namespaces, err := getNamespaces(client, namespacePtr)
 	if err != nil {
 		log.Fatalf("error getting namespaces: %v", err)
+	}
+	namespaces, err = filterNamespaces(namespaces, namespacePatternPtr)
+	if err != nil {
+		log.Fatalf("error filtering namespaces with pattern %s: %v", *namespacePatternPtr, err)
 	}
 	for _, n := range namespaces {
 
@@ -109,8 +115,6 @@ func getNamespaces(client *kubernetes.Clientset, namespaceFlag *string) ([]strin
 		return strings.Split(*namespaceFlag, ","), nil
 	}
 
-	log.Println("generating report for all namespaces")
-
 	namespaceList, err := client.CoreV1().Namespaces().List(metav1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("error querying namespaces: %w", err)
@@ -119,6 +123,35 @@ func getNamespaces(client *kubernetes.Clientset, namespaceFlag *string) ([]strin
 	result := make([]string, len(namespaceList.Items))
 	for i, ns := range namespaceList.Items {
 		result[i] = ns.Name
+	}
+
+	return result, nil
+}
+
+// filterNamespaces returns a slice of strings from the `namespaces` parameter
+// that match the `namespacePatternFlag` regex. If `namespacePatternFlag` has
+// no value, all namespaces are included.
+func filterNamespaces(namespaces []string, namespacePatternFlag *string) ([]string, error) {
+	if namespacePatternFlag == nil || *namespacePatternFlag == "" {
+		return namespaces, nil
+	}
+
+	if len(namespaces) == 0 {
+		return namespaces, nil
+	}
+
+	pattern := strings.Trim(*namespacePatternFlag, "\"")
+
+	result := make([]string, 0, len(namespaces))
+	re, err := regexp.Compile(pattern)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, namespace := range namespaces {
+		if re.MatchString(namespace) {
+			result = append(result, namespace)
+		}
 	}
 
 	return result, nil

--- a/pkg/cluster/images.go
+++ b/pkg/cluster/images.go
@@ -2,6 +2,9 @@ package cluster
 
 import (
 	"fmt"
+	"regexp"
+	"strings"
+
 	"github.com/integr8ly/heimdall/pkg/domain"
 	v12 "github.com/openshift/api/apps/v1"
 	v13 "github.com/openshift/api/image/v1"
@@ -9,9 +12,7 @@ import (
 	"github.com/pkg/errors"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
-	"regexp"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
-	"strings"
 )
 
 var log = logf.Log.WithName("image_service")
@@ -182,7 +183,8 @@ func ParseImage(i string) *domain.ClusterImage {
 	registryURLParts := strings.Split(si, "/")
 	imageParts := strings.Split(registryURLParts[len(registryURLParts)-1], ":")
 	registryURL := strings.Join(registryURLParts[:len(registryURLParts)-1], "/") + "/" + imageParts[0]
-	imagePath := strings.Split(strings.Split(si, ":")[0], "/")
+	imagePath := strings.Split(si, "/")
+	imagePath[2] = strings.Split(imagePath[2], ":")[0] // remove the tag from the last part
 	if len(imageParts) == 1 {
 		imageParts = append(imageParts, "latest")
 	}

